### DISCUSSION TITLE
Persistance setting to actually open apps

### DISCUSSION THREAD

**Discussion by SavuBogdan** [material-shell/material-shell#760] [Ideas]
Hello would it be possible to have an option where the environment and workspaces is fully restarted and apps are opened instead of just shortcuts to what was open before ?

Also a way to save the layout instead of remembering the last state since most of the time you have a prefered layout and if you deviate to much from it you could have a quick restore if you reboot

**Comment by PapyElGringo** [comment]
Well it should be possible but I think it's would be less convenient since most if the people doesn't use all applications at once and instead have some workspace dedicated to different activities and start all of them would be unefficient for the computer.

You can mark windows as fully persistent which mean that even if you close them they will stay where you want them to be. This way if you close all workspaces (with middle click on them) you will retrieve only the fully persistent windows

**Comment by Luctins** [comment]
You could also set them to start on boot, but then the problem is to have to maintain two separate lists of applications. Would it be hard to make some integration?

**Comment by seantcanavan** [comment]
This would be an amazing feature and ridiculously helpful @PapyElGringo. I just switched to Manjaro on all 4 of my machines and fell in love with Material Shell instantly and have it configured on all 4. All 4 machines are powerful enough to launch their 4-6 programs on startup and it would be awesome if material shell did this instead of me clicking through each workspace and clicking on each tile to launch it. Something like:

"Auto start every application in every workspace on material shell startup. WARNING: This is only recommended for high powered machines".

Another option if you're worried about performance is a configurable number of workspaces to autostart on startup. So if someone were to have 10-20 workspaces the setting wouldn't cascade and then crush the CPU.
